paparazzi/src/main/java/app/cash/paparazzi/internal/resources/PseudolocaleGenerator.kt

```diff
@@ -0,0 +1,144 @@
+        if (value !is BasicValueResourceItemBase || !value.isPseudolocalizable()) {
+          return@forEach
+        }
+
+        val method = when (localeQualifier.value) {
+          "en-rXA" -> Pseudolocalizer.Method.ACCENT
+          "ar-rXB" -> Pseudolocalizer.Method.BIDI
+          else -> return@forEach
+        }
+
+        var repositoryConfiguration = configCache[value.sourceFile to localeQualifier]
+        if (repositoryConfiguration == null) {
+          repositoryConfiguration = value.sourceFile.configuration.onNewLocaleQualifier(localeQualifier)
+          configCache[value.sourceFile to localeQualifier] = repositoryConfiguration
+        }
+
+        val pseudoLocaleSourceFile = value.sourceFile.onNewRepositoryConfiguration(repositoryConfiguration)
+        val pseudoItem: BasicValueResourceItemBase = when (value.resourceType) {
+          ResourceType.STRING -> pseudolocalizeString(
+            value as BasicValueResourceItem,
+            pseudoLocaleSourceFile,
+            method
+          )
+
+          ResourceType.PLURALS -> pseudolocalizePlural(
+            value as BasicPluralsResourceItem,
+            pseudoLocaleSourceFile,
+            method
+          )
+
+          ResourceType.ARRAY -> pseudolocalizeArray(
+            value as BasicArrayResourceItem,
+            pseudoLocaleSourceFile,
+            method
+          )
+
+          else -> return@forEach
+        }
+
+        resourceValues[key] = pseudoItem
+      }
+    }
+  }
+
+  return this
+}
+
+/**
+ * Check if the resource item is pseudolocalizable by verifying the source file name.
+ *
+ * Currently we cannot verify the translatable attribute from raw xml as it is not stored in BasicValueResourceItemBase.
+ * This will cause an issue that untranslatable strings are translated. The solution will be a new
+ * translatable Boolean field in BasicValueResourceItemBase, but this requires changes in Android Open Source Project.
+ */
+private fun BasicValueResourceItemBase.isPseudolocalizable(): Boolean =
+  sourceFile.configuration.folderConfiguration.localeQualifier == null && sourceFile.relativePath?.contains(DO_NOT_TRANSLATE) != true
+
+private fun RepositoryConfiguration.onNewLocaleQualifier(localeQualifier: LocaleQualifier): RepositoryConfiguration {
+  val folderConfiguration = FolderConfiguration.copyOf(folderConfiguration)
+  folderConfiguration.localeQualifier = localeQualifier
+  return RepositoryConfiguration(repository, folderConfiguration)
+}
+
+private fun ResourceSourceFile.onNewRepositoryConfiguration(configuration: RepositoryConfiguration): ResourceSourceFile {
+  return when (this) {
+    is ResourceSourceFileImpl -> ResourceSourceFileImpl(relativePath, configuration)
+    is ResourceFile -> ResourceFile(file, configuration)
+    // exhaustive to avoid potential breaking changes in future releases
+    else -> throw IllegalStateException("${javaClass.name} is not handled for new RepositoryConfiguration")
+  }
+}
```

Comment:
```suggestion
private fun ResourceSourceFile.forLocale(localeQualifier: LocaleQualifier): ResourceSourceFile {
  val repositoryConfiguration = configCache.getOrPut(this to localeQualifier) {
    RepositoryConfiguration(
      repository = configuration.repository,
      folderConfiguration = FolderConfiguration.copyOf(configuration.folderConfiguration).apply {
        this.localeQualifier = localeQualifier
      }
    )
  }
  return when (this) {
    is ResourceSourceFileImpl -> ResourceSourceFileImpl(relativePath, repositoryConfiguration)
    is ResourceFile -> ResourceFile(file, repositoryConfiguration)
    // exhaustive to avoid potential breaking changes in future releases
    else -> throw IllegalStateException("${javaClass.name} is not handled for new RepositoryConfiguration")
  }
}
```